Sources/NIOWebSocket/WebSocketErrorCodes.swift

```diff
@@ -0,0 +1,214 @@
+/// to avoid users needing to remember the specific numerical values
+/// of those codes.
+public enum WebSocketErrorCode {
+    /// Indicates a normal closure, meaning that the purpose for
+    /// which the connection was established has been fulfilled.
+    /// Corresponds to code 1000.
+    case normalClosure
+
+    /// Ondicates that an endpoint is "going away", such as a server
+    /// going down or a browser having navigated away from a page.
+    /// Corresponds to code 1001.
+    case goingAway
+
+    /// Indicates that an endpoint is terminating the connection due
+    /// to a protocol error.
+    /// Corresponds to code 1002.
+    case protocolError
+
+    /// Indicates that an endpoint is terminating the connection
+    /// because it has received a type of data it cannot accept (e.g. an
+    /// endpoint that understands only text data may send this if it
+    /// receives a binary message).
+    /// Corresponds to code 1003.
+    case unacceptableData
+
+    /// Indicates that an endpoint is terminating the connection
+    /// because it has received data within a message that was not
+    /// consistent with the type of the message (e.g. non-UTF-8
+    /// data within a text message).
+    /// Corresponds to code 1007.
+    case dataInconsistentWithMessage
+
+    /// Indicates that an endpoint is terminating the connection
+    /// because it has received a message that violates its policy.  This
+    /// is a generic status code that can be returned when there is no
+    /// other more suitable status code (e.g. 1003 or 1009) or if there
+    /// is a need to hide specific details about the policy.
+    /// Corresponds to code 1008.
+    case policyViolation
+
+    /// Indicates that an endpoint is terminating the connection
+    /// because it has received a message that is too big for it to
+    /// process.
+    /// Corresponds to code 1009.
+    case messageTooLarge
+
+    /// Indicates that an endpoint (client) is terminating the
+    /// connection because it has expected the server to negotiate one or
+    /// more extension, but the server didn't return them in the response
+    /// message of the WebSocket handshake.  The list of extensions that
+    /// are needed should appear in the `reason` part of the Close frame.
+    /// Note that this status code is not used by the server, because it
+    /// can fail the WebSocket handshake instead.
+    /// Corresponds to code 1010.
+    case missingExtension
+
+    /// Indicates that a server is terminating the connection because
+    /// it encountered an unexpected condition that prevented it from
+    /// fulfilling the request.
+    /// Corresponds to code 1011.
+    case unexpectedServerError
+
+    /// We don't have a better name for this error code.
+    case unknown(UInt16)
+
+    /// Create an error code from a raw 16-bit integer.
+    ///
+    /// - parameters:
+    ///     integer: The integer form of the status code.
+    public init(_ integer: UInt16) {
```

Comment:
I think this should have a label like `webSocketsErrorCode` and should be `Int`, no?

Reply:
We have one that takes `Int`: the one that takes `UInt16` is used in the protocol parsing. If we remove it, we have to uplift to `Int` in the protocol parsing instead. Not sure we gain much by that? Up to you though, I'll defer if you think we should have only the `Int` one.

Reply:
fair enough, let's have both but with labels. Potentially the `UInt16` one could be internal or smth